notebooks/linear_regression.livemd

```diff
@@ -0,0 +1,610 @@
+# Linear Regression in Practice
+
+```elixir
+Mix.install(
+  [
+    {:scholar, github: "elixir-nx/scholar"},
+    {:explorer, github: "elixir-nx/explorer"},
+    {:nx, github: "elixir-nx/nx", sparse: "nx", override: true},
+    {:exla, github: "elixir-nx/nx", sparse: "exla", override: true},
+    {:rustler, ">= 0.0.0", optional: true},
+    {:req, "~> 0.3.3"},
+    {:vega_lite, "~> 0.1.6"},
+    {:kino_vega_lite, "~> 0.1.7"},
+    {:kino, "~> 0.8.0"}
+  ],
+  config: [
+    nx: [default_defn_options: [compiler: EXLA]]
+  ]
+)
+```
+
+## Introduction
+
+In the livebook, we will cover the typical use cases of linear regression on practical examples.
+
+```elixir
+alias VegaLite, as: Vl
+require Explorer.DataFrame
+```
+
+<!-- livebook:{"branch_parent_index":0} -->
+
+## Linear Regression on Synthetic Data
+
+```elixir
+use Nx.Defn.Kernel
```

Comment:
I'm not sure if it's a good idea to actually `use Nx.Defn.Kernel` directly here. It has `@doc false`, which makes me think it's not actually a public API in itself.

@josevalim WDYT?

Reply:
@msluszniak last thing here. I confirmed that this is a private API, so we need to remove this `use`. I think this only affects the next cell, right?

Perhaps just adding a module around it will suffice

Reply:
Sure